src/iris/api.py

```diff
@@ -2514,12 +2520,30 @@ def on_put(self, req, resp, app_name):
                     {'application_id': app['id']})
             }
 
+            title_variable = data.get('title_variable')
+            # if no variables are set then title variable will be null
+            if not new_variables:
+                title_variable = None
+            elif title_variable and title_variable not in new_variables:
+                raise HTTPBadRequest('title variable is invalid')
+
             kill_variables = existing_variables - new_variables
 
-            for variable in new_variables - existing_variables:
-                session.execute('''INSERT INTO `template_variable` (`application_id`, `name`)
-                                VALUES (:application_id, :variable)''',
-                                {'application_id': app['id'], 'variable': variable})
+            # insert new variables and update the value of title_variable for existing variables
```

Comment:
We can reduce DB queries by keeping the old variable insertion query, then doing a `UPDATE template_variable SET title_variable=(CASE WHEN name=:title_val THEN TRUE ELSE FALSE) WHERE application_id = (SELECT id from application where name=:app)` (or something similar, I didn't test this).

Reply:
That makes sense, I'll make a change to that effect.